be/src/exec/vectorized/cross_join_node.cpp

```diff
@@ -537,33 +537,31 @@ void CrossJoinNode::_init_chunk(ChunkPtr* chunk) {
 pipeline::OpFactories CrossJoinNode::decompose_to_pipeline(pipeline::PipelineBuilderContext* context) {
     using namespace pipeline;
 
-    std::shared_ptr<pipeline::CrossJoinContext> cross_join_context = std::make_shared<pipeline::CrossJoinContext>();
-
     // step 0: construct pipeline end with cross join right operator.
-    OpFactories operator_before_cross_join_right = _children[1]->decompose_to_pipeline(context);
-    operator_before_cross_join_right =
-            context->maybe_interpolate_local_passthrough_exchange(operator_before_cross_join_right);
+    OpFactories right_ops = _children[1]->decompose_to_pipeline(context);
 
     // communication with CrossJoinLeft through shared_datas.
+    auto* right_source = down_cast<SourceOperatorFactory*>(right_ops[0].get());
+    std::shared_ptr<pipeline::CrossJoinContext> cross_join_context =
```

Comment:
Maybe It's better to use **auto cross_join_context**, It's better to keep consistent with others.